test/client/autosuggest.spec.ts

```diff
@@ -0,0 +1,532 @@
+    const clipToPolygon = [
+      startEndCoordinate,
+      generateCoordinate(),
+      generateCoordinate(),
+      generateCoordinate(),
+      startEndCoordinate,
+    ];
+    const language = CHANCE.locale();
+    const preferLand = CHANCE.bool();
+    const transportArguments = {
+      method: 'get',
+      host: `${host.replace(/\/$/, '')}/${apiVersion}`,
+      url: '/autosuggest',
+      query: {
+        key: apiKey,
+        input,
+        'n-results': `${nResults}`,
+        focus: `${focus.lat},${focus.lng}`,
+        'n-focus-results': `${nFocusResults}`,
+        'clip-to-bounding-box': `${clipToBoundingBox.southwest.lat},${clipToBoundingBox.southwest.lng},${clipToBoundingBox.northeast.lat},${clipToBoundingBox.northeast.lng}`,
+        'clip-to-circle': `${clipToCircle.center.lat},${clipToCircle.center.lng},${clipToCircle.radius}`,
+        'clip-to-country': clipToCountry.join(','),
+        'clip-to-polygon': clipToPolygon
+          .map(coord => `${coord.lat},${coord.lng}`)
+          .join(','),
+        language,
+        'prefer-land': `${preferLand}`,
+        'input-type': inputType,
+      },
+      headers: { 'X-Api-Key': apiKey, ...HEADERS },
+      body: null,
+    };
+
+    await client.run({
+      input,
+      inputType,
+      nResults,
+      nFocusResults,
+      focus,
+      clipToBoundingBox,
+      clipToCircle,
+      clipToCountry,
+      clipToPolygon,
+      language,
+      preferLand,
+    });
+    transportSpy
+      .calledOnceWith(transportArguments)
+      .should.be.equal(true, 'transport arguments do not match');
+  });
+  it('should call /autosuggest with voice input type', async () => {
+    const input = `${CHANCE.word()}.${CHANCE.word()}.${CHANCE.letter()}`;
+    const inputType = CHANCE.pickone([
+      AutosuggestInputType.VoconHybrid,
+      AutosuggestInputType.NMDP_ASR,
+      AutosuggestInputType.GenericVoice,
+    ]);
+    const language = CHANCE.locale();
+    const transportArguments = {
+      method: 'get',
+      host: `${host.replace(/\/$/, '')}/${apiVersion}`,
+      url: '/autosuggest',
+      query: {
+        key: apiKey,
+        input,
+        'input-type': inputType,
+        language,
+      },
+      headers: { 'X-Api-Key': apiKey, ...HEADERS },
+      body: null,
+    };
+
+    await client.run({
+      input,
+      inputType,
+      language,
+    });
+    transportSpy
+      .calledOnceWith(transportArguments)
+      .should.be.equal(true, 'transport arguments do not match');
+  });
+  it('should throw error when no language provided with voice input type', async () => {
+    const input = `${CHANCE.word()}.${CHANCE.word()}.${CHANCE.letter()}`;
+    const inputType = CHANCE.pickone([
+      AutosuggestInputType.VoconHybrid,
+      AutosuggestInputType.NMDP_ASR,
+      AutosuggestInputType.GenericVoice,
+    ]);
+
+    try {
+      await client.run({
+        input,
+        inputType,
+      });
+    } catch (err) {
+      err.message.should.be.equal(
```

Comment:
For error tests you should assert the number of assertions (meta) in the finally block. If something went wrong and the error wasn't thrown then the tests might not otherwise pick up on it otherwise.

Reply:
The test will fail if the error is not thrown as the `...finally` checks that the `spy` was not called. If the `spy` was called it means an error was not thrown.

Reply:
Sorry I didn't realise that was checking the same thing, my bad. All good!